api/v1alpha1/accesslogging_types.go

```diff
@@ -86,6 +96,60 @@ type ProxyAccessLogSink struct {
 	OpenTelemetry *OpenTelemetryEnvoyProxyAccessLog `json:"openTelemetry,omitempty"`
 }
 
+type ALSEnvoyProxyAccessLogType string
+
+const (
+	// ALSEnvoyProxyAccessLogTypeHTTP defines the HTTP access log type and will populate StreamAccessLogsMessage.http_logs.
+	ALSEnvoyProxyAccessLogTypeHTTP ALSEnvoyProxyAccessLogType = "HTTP"
+	// ALSEnvoyProxyAccessLogTypeTCP defines the TCP access log type and will populate StreamAccessLogsMessage.tcp_logs.
+	ALSEnvoyProxyAccessLogTypeTCP ALSEnvoyProxyAccessLogType = "TCP"
+)
+
+// ALSEnvoyProxyAccessLog defines the gRPC Access Log Service (ALS) sink.
+// The service must implement the Envoy gRPC Access Log Service streaming API:
+// https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/accesslog/v3/als.proto
+// Access log format information is passed in the form of gRPC metadata when the
+// stream is established. Specifically, the following metadata is passed:
+//
+// - `x-accesslog-text` - The access log format string when a Text format is used.
+//
+// - `x-accesslog-attr` - JSON encoded key/value pairs when a JSON format is used.
+//
+// +kubebuilder:validation:XValidation:message="BackendRef only supports Service Kind.",rule="!has(self.backendRef.kind) || self.backendRef.kind == 'Service'"
+// +kubebuilder:validation:XValidation:rule="self.type == 'HTTP' || !has(self.http)",message="The http field may only be set when type is HTTP."
+type ALSEnvoyProxyAccessLog struct {
+	// BackendRef references a Kubernetes object that represents the gRPC service to which
+	// the access logs will be sent. Currently only Service is supported.
+	BackendRef gwapiv1.BackendObjectReference `json:"backendRef,omitempty"`
+	// LogName defines the friendly name of the access log to be returned in
+	// StreamAccessLogsMessage.Identifier. This allows the access log server
+	// to differentiate between different access logs coming from the same Envoy.
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:default="accesslog"
+	LogName string `json:"logName,omitempty"`
+	// Type defines the type of accesslog. Supported types are "HTTP" and "TCP". Defaults to "HTTP" when not specified.
+	// +kubebuilder:validation:Enum=HTTP;TCP
+	// +unionDiscriminator
+	// +kubebuilder:default="HTTP"
+	Type ALSEnvoyProxyAccessLogType `json:"type,omitempty"`
```

Comment:
only suppot HTTP right now?

Reply:
since there's no config specific to TCP — basically just a different type — I figured we may as well include both in the API design. although we could easily pull the `type` field out entirely until a later date if it'd be more amenable to keep the API specific to HTTP for now.